### PR TITLE
Change the default for pub tracing to false for non-bot runs

### LINKED ISSUE #13179: flutter packages get is always verbose
If `flutter packages get` fails -- for example on a bad `path:` in a dependency -- it prints way to much output, drowning the actionable information at the top.

I suggest we only print the log transcript when the `-v` option is used.

```
mit-macbookpro2:authtest mit$ flutter packages get
Running "flutter packages get" in authtest...
Could not find package firebase_auth at "../packages/firebase_auth".
Depended on by:
- authtest
---- Log transcript ----
FINE: Pub 1.25.0-dev.11.0
MSG : Resolving dependencies...
SLVR: Solving dependencies:
    | - firebase_auth any from path ({path: ../packages/firebase_auth, relative: true})
    | - cupertino_icons ^0.1.0 from hosted (cupertino_icons) (locked to 0.1.1)
    | - flutter any from sdk (flutter) (locked to 0.0.38)
SLVR: Could not get versions for firebase_auth from path:
    | Could not find package firebase_auth at "../packages/firebase_auth".
    |
    | package:pub/src/source/path.dart 177                                          BoundPathSource._validatePath
    | package:pub/src/source/path.dart 147                                          BoundPathSource._loadPubspec
    | package:pub/src/source/path.dart 138                                          BoundPathSource.doGetVersions
    | package:pub/src/source.dart 165                                               BoundSource.getVersions
    | package:pub/src/solver/version_solver.dart 231                                SolverCache.getVersions.<fn>
    | dart:async                                                                    runZoned
    | package:pub/src/http.dart 228                                                 withDependencyType
    | package:pub/src/solver/version_solver.dart 230                                SolverCache.getVersions
    | package:pub/src/solver/unselected_package_queue.dart 121                      UnselectedPackageQueue._getNumVersions
    | package:pub/src/solver/unselected_package_queue.dart 50                       UnselectedPackageQueue.add
    | package:pub/src/solver/version_selection.dart 68                              VersionSelection.select.<fn>
    | dart:async                                                                    Future.forEach
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select
    | package:pub/src/solver/backtracking_solver.dart 172                           BacktrackingSolver.solve
    | package:pub/src/solver/version_solver.dart 42                                 resolveVersions.<fn>
    | package:pub/src/log.dart 409                                                  progress
    | package:pub/src/solver/version_solver.dart 40                                 resolveVersions
    | package:pub/src/entrypoint.dart 193                                           Entrypoint.acquireDependencies
    | package:pub/src/command/get.dart 38                                           GetCommand.run
    | package:args/command_runner.dart 194                                          CommandRunner.runCommand
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand.<fn>
    | dart:async                                                                    new Future.sync
    | package:pub/src/utils.dart 102                                                captureErrors.<fn>
    | package:stack_trace                                                           Chain.capture
    | package:pub/src/utils.dart 117                                                captureErrors
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand
    | package:pub/src/command_runner.dart 117                                       PubCommandRunner.run
    | /b/build/slave/dart-sdk-mac-dev/build/sdk/third_party/pkg/pub/bin/pub.dart 8  main
    | ===== asynchronous gap ===========================
    | dart:async                                                                    _Completer.completeError
    | package:pub/src/source/path.dart 142                                          BoundPathSource.doGetVersions
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/source/path.dart 135                                          BoundPathSource.doGetVersions
    | package:pub/src/source.dart 165                                               BoundSource.getVersions
    | package:pub/src/solver/version_solver.dart 231                                SolverCache.getVersions.<fn>
    | dart:async                                                                    runZoned
    | package:pub/src/http.dart 228                                                 withDependencyType
    | package:pub/src/solver/version_solver.dart 230                                SolverCache.getVersions
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/version_solver.dart 204                                SolverCache.getVersions
    | package:pub/src/solver/unselected_package_queue.dart 121                      UnselectedPackageQueue._getNumVersions
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/unselected_package_queue.dart 115                      UnselectedPackageQueue._getNumVersions
    | package:pub/src/solver/unselected_package_queue.dart 50                       UnselectedPackageQueue.add
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/unselected_package_queue.dart 44                       UnselectedPackageQueue.add
    | package:pub/src/solver/version_selection.dart 68                              VersionSelection.select.<fn>
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select.<fn>
    | dart:async                                                                    Future.forEach
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select
    | ===== asynchronous gap ===========================
    | dart:async                                                                    Future.forEach
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select
    | ===== asynchronous gap ===========================
    | dart:async                                                                    _asyncThenWrapperHelper
    | package:pub/src/solver/version_selection.dart 53                              VersionSelection.select
    | package:pub/src/solver/backtracking_solver.dart 172                           BacktrackingSolver.solve
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/backtracking_solver.dart 158                           BacktrackingSolver.solve
    | package:pub/src/solver/version_solver.dart 42                                 resolveVersions.<fn>
    | package:pub/src/log.dart 409                                                  progress
    | package:pub/src/solver/version_solver.dart 40                                 resolveVersions
    | package:pub/src/entrypoint.dart 193                                           Entrypoint.acquireDependencies
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/entrypoint.dart 192                                           Entrypoint.acquireDependencies
    | package:pub/src/command/get.dart 38                                           GetCommand.run
    | package:args/command_runner.dart 194                                          CommandRunner.runCommand
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:args/command_runner.dart 142                                          CommandRunner.runCommand
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand.<fn>
    | dart:async                                                                    new Future.sync
    | package:pub/src/utils.dart 102                                                captureErrors.<fn>
    | package:stack_trace                                                           Chain.capture
    | package:pub/src/utils.dart 117                                                captureErrors
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand
SLVR: Could not get versions for firebase_auth from path:
    | Could not find package firebase_auth at "../packages/firebase_auth".
    |
    | package:pub/src/source/path.dart 177                                          BoundPathSource._validatePath
    | package:pub/src/source/path.dart 147                                          BoundPathSource._loadPubspec
    | package:pub/src/source/path.dart 138                                          BoundPathSource.doGetVersions
    | package:pub/src/source.dart 165                                               BoundSource.getVersions
    | package:pub/src/solver/version_solver.dart 231                                SolverCache.getVersions.<fn>
    | dart:async                                                                    runZoned
    | package:pub/src/http.dart 228                                                 withDependencyType
    | package:pub/src/solver/version_solver.dart 230                                SolverCache.getVersions
    | package:pub/src/solver/unselected_package_queue.dart 121                      UnselectedPackageQueue._getNumVersions
    | package:pub/src/solver/unselected_package_queue.dart 50                       UnselectedPackageQueue.add
    | package:pub/src/solver/version_selection.dart 68                              VersionSelection.select.<fn>
    | dart:async                                                                    Future.forEach
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select
    | package:pub/src/solver/backtracking_solver.dart 172                           BacktrackingSolver.solve
    | package:pub/src/solver/version_solver.dart 42                                 resolveVersions.<fn>
    | package:pub/src/log.dart 409                                                  progress
    | package:pub/src/solver/version_solver.dart 40                                 resolveVersions
    | package:pub/src/entrypoint.dart 193                                           Entrypoint.acquireDependencies
    | package:pub/src/command/get.dart 38                                           GetCommand.run
    | package:args/command_runner.dart 194                                          CommandRunner.runCommand
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand.<fn>
    | dart:async                                                                    new Future.sync
    | package:pub/src/utils.dart 102                                                captureErrors.<fn>
    | package:stack_trace                                                           Chain.capture
    | package:pub/src/utils.dart 117                                                captureErrors
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand
    | package:pub/src/command_runner.dart 117                                       PubCommandRunner.run
    | /b/build/slave/dart-sdk-mac-dev/build/sdk/third_party/pkg/pub/bin/pub.dart 8  main
    | ===== asynchronous gap ===========================
    | dart:async                                                                    _Completer.completeError
    | package:pub/src/source/path.dart 142                                          BoundPathSource.doGetVersions
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/source/path.dart 135                                          BoundPathSource.doGetVersions
    | package:pub/src/source.dart 165                                               BoundSource.getVersions
    | package:pub/src/solver/version_solver.dart 231                                SolverCache.getVersions.<fn>
    | dart:async                                                                    runZoned
    | package:pub/src/http.dart 228                                                 withDependencyType
    | package:pub/src/solver/version_solver.dart 230                                SolverCache.getVersions
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/version_solver.dart 204                                SolverCache.getVersions
    | package:pub/src/solver/unselected_package_queue.dart 121                      UnselectedPackageQueue._getNumVersions
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/unselected_package_queue.dart 115                      UnselectedPackageQueue._getNumVersions
    | package:pub/src/solver/unselected_package_queue.dart 50                       UnselectedPackageQueue.add
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/unselected_package_queue.dart 44                       UnselectedPackageQueue.add
    | package:pub/src/solver/version_selection.dart 68                              VersionSelection.select.<fn>
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select.<fn>
    | dart:async                                                                    Future.forEach
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select
    | ===== asynchronous gap ===========================
    | dart:async                                                                    Future.forEach
    | package:pub/src/solver/version_selection.dart 61                              VersionSelection.select
    | ===== asynchronous gap ===========================
    | dart:async                                                                    _asyncThenWrapperHelper
    | package:pub/src/solver/version_selection.dart 53                              VersionSelection.select
    | package:pub/src/solver/backtracking_solver.dart 172                           BacktrackingSolver.solve
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/solver/backtracking_solver.dart 158                           BacktrackingSolver.solve
    | package:pub/src/solver/version_solver.dart 42                                 resolveVersions.<fn>
    | package:pub/src/log.dart 409                                                  progress
    | package:pub/src/solver/version_solver.dart 40                                 resolveVersions
    | package:pub/src/entrypoint.dart 193                                           Entrypoint.acquireDependencies
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:pub/src/entrypoint.dart 192                                           Entrypoint.acquireDependencies
    | package:pub/src/command/get.dart 38                                           GetCommand.run
    | package:args/command_runner.dart 194                                          CommandRunner.runCommand
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:args/command_runner.dart 142                                          CommandRunner.runCommand
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand.<fn>
    | dart:async                                                                    new Future.sync
    | package:pub/src/utils.dart 102                                                captureErrors.<fn>
    | package:stack_trace                                                           Chain.capture
    | package:pub/src/utils.dart 117                                                captureErrors
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand
SLVR: * start at root
SLVR: | cupertino_icons 0.1.1 from hosted is locked
SLVR: | * select cupertino_icons 0.1.1 from hosted
SLVR: | | flutter 0.0.38 from sdk is locked
SLVR: | | * select flutter 0.0.38 from sdk
SLVR: | | | collection 1.14.3 from hosted is locked
SLVR: | | | * select collection 1.14.3 from hosted
SLVR: | | | | flutter_test 0.0.16 from sdk is locked
SLVR: | | | | * select flutter_test 0.0.16 from sdk
SLVR: | | | | | * select pub itself
SLVR: | | | | | | args 0.13.7 from hosted is locked
SLVR: | | | | | | * select args 0.13.7 from hosted
SLVR: | | | | | | | async 1.13.3 from hosted is locked
SLVR: | | | | | | | * select async 1.13.3 from hosted
SLVR: | | | | | | | | barback 0.15.2+13 from hosted is locked
SLVR: | | | | | | | | * select barback 0.15.2+13 from hosted
SLVR: | | | | | | | | | boolean_selector 1.0.2 from hosted is locked
SLVR: | | | | | | | | | * select boolean_selector 1.0.2 from hosted
SLVR: | | | | | | | | | | charcode 1.1.1 from hosted is locked
SLVR: | | | | | | | | | | * select charcode 1.1.1 from hosted
SLVR: | | | | | | | | | | | convert 2.0.1 from hosted is locked
SLVR: | | | | | | | | | | | * select convert 2.0.1 from hosted
SLVR: | | | | | | | | | | | | crypto 2.0.2+1 from hosted is locked
SLVR: | | | | | | | | | | | | * select crypto 2.0.2+1 from hosted
SLVR: | | | | | | | | | | | | | glob 1.1.5 from hosted is locked
SLVR: | | | | | | | | | | | | | * select glob 1.1.5 from hosted
SLVR: | | | | | | | | | | | | | | http 0.11.3+14 from hosted is locked
SLVR: | | | | | | | | | | | | | | * select http 0.11.3+14 from hosted
SLVR: | | | | | | | | | | | | | | | http_multi_server 2.0.4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | * select http_multi_server 2.0.4 from hosted
SLVR: | | | | | | | | | | | | | | | | http_parser 3.1.1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | * select http_parser 3.1.1 from hosted
SLVR: | | | | | | | | | | | | | | | | | io 0.3.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | * select io 0.3.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | js 0.6.1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | * select js 0.6.1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | matcher 0.12.1+4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | * select matcher 0.12.1+4 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | meta 1.1.1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | * select meta 1.1.1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | mime 0.9.4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | * select mime 0.9.4 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | mockito 2.2.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | * select mockito 2.2.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | node_preamble 1.4.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | * select node_preamble 1.4.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | package_config 1.0.3 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | * select package_config 1.0.3 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | package_resolver 1.0.2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | * select package_resolver 1.0.2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | path 1.4.2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | * select path 1.4.2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | pool 1.3.3 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | * select pool 1.3.3 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | pub_semver 1.3.2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select pub_semver 1.3.2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | quiver 0.26.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select quiver 0.26.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | shelf 0.7.1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select shelf 0.7.1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | shelf_packages_handler 1.0.3 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select shelf_packages_handler 1.0.3 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | shelf_static 0.2.6 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select shelf_static 0.2.6 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | shelf_web_socket 0.2.2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select shelf_web_socket 0.2.2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | sky_engine 0.0.99 from path is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select sky_engine 0.0.99 from path
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | source_map_stack_trace 1.1.4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select source_map_stack_trace 1.1.4 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | source_maps 0.10.4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select source_maps 0.10.4 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | source_span 1.4.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select source_span 1.4.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | stack_trace 1.9.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select stack_trace 1.9.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | stream_channel 1.6.2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select stream_channel 1.6.2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | string_scanner 1.0.2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select string_scanner 1.0.2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | term_glyph 1.0.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select term_glyph 1.0.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | test 0.12.26 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select test 0.12.26 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | analyzer 0.30.0+4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select analyzer 0.30.0+4 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | cli_util 0.1.2+1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select cli_util 0.1.2+1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | front_end 0.1.0-alpha.4.1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select front_end 0.1.0-alpha.4.1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | html 0.13.2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select html 0.13.2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | csslib 0.14.1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select csslib 0.14.1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | isolate 1.1.0 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select isolate 1.1.0 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | kernel 0.3.0-alpha.1.1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select kernel 0.3.0-alpha.1.1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | logging 0.11.3+1 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select logging 0.11.3+1 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | plugin 0.2.0+2 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select plugin 0.2.0+2 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | typed_data 1.1.4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select typed_data 1.1.4 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | utf 0.9.0+3 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select utf 0.9.0+3 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | vector_math 2.0.5 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select vector_math 2.0.5 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | watcher 0.9.7+4 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select watcher 0.9.7+4 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | web_socket_channel 1.0.6 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select web_socket_channel 1.0.6 from hosted
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | yaml 2.1.13 from hosted is locked
SLVR: | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | | * select yaml 2.1.13 from hosted
SLVR: BacktrackingSolver took 0:00:00.306855 seconds.
    | - Tried 1 solutions
    | - Requested 1 version lists
    | - Looked up 1 cached version lists
    |
FINE: Resolving dependencies finished (0.3s).
ERR : Could not find package firebase_auth at "../packages/firebase_auth".
    | Depended on by:
    | - authtest
FINE: Exception type: DependencyNotFoundException
FINE: package:pub/src/entrypoint.dart 195                                           Entrypoint.acquireDependencies
    | package:pub/src/command/get.dart 38                                           GetCommand.run
    | package:args/command_runner.dart 194                                          CommandRunner.runCommand
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand.<fn>
    | dart:async                                                                    new Future.sync
    | package:pub/src/utils.dart 102                                                captureErrors.<fn>
    | package:stack_trace                                                           Chain.capture
    | package:pub/src/utils.dart 117                                                captureErrors
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand
    | package:pub/src/command_runner.dart 117                                       PubCommandRunner.run
    | /b/build/slave/dart-sdk-mac-dev/build/sdk/third_party/pkg/pub/bin/pub.dart 8  main
    | ===== asynchronous gap ===========================
    | dart:async                                                                    _Completer.completeError
    | package:pub/src/entrypoint.dart 243                                           Entrypoint.acquireDependencies
    | ===== asynchronous gap ===========================
    | dart:async                                                                    _asyncThenWrapperHelper
    | package:pub/src/entrypoint.dart 192                                           Entrypoint.acquireDependencies
    | package:pub/src/command/get.dart 38                                           GetCommand.run
    | package:args/command_runner.dart 194                                          CommandRunner.runCommand
    | ===== asynchronous gap ===========================
    | dart:async                                                                    new Future.microtask
    | package:args/command_runner.dart 142                                          CommandRunner.runCommand
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand.<fn>
    | dart:async                                                                    new Future.sync
    | package:pub/src/utils.dart 102                                                captureErrors.<fn>
    | package:stack_trace                                                           Chain.capture
    | package:pub/src/utils.dart 117                                                captureErrors
    | package:pub/src/command_runner.dart 168                                       PubCommandRunner.runCommand
---- End log transcript ----
pub get failed (69) -- attempting retry 1 in 1 second...
```

### PR DESCRIPTION
**Pull request by mit-mit**
Fixes #13179

### CODE REVIEW

File: packages/flutter_tools/lib/src/dart/pub.dart [hunk 1]
```diff
@@ -134,8 +134,10 @@ Future<Null> pub(List<String> arguments, {
   MessageFilter filter,
   String failureMessage: 'pub failed',
   @required bool retry,
-  bool showTraceForErrors: true,
+  bool showTraceForErrors,
 }) async {
+  showTraceForErrors ??= isRunningOnBot;
+
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/dart/pub.dart]: we should also update the docs

Comment (mit-mit) [reply packages/flutter_tools/lib/src/dart/pub.dart]: Done, PTAL

**Comment by mit-mit** [comment]
cc @Hixie can you take another look at this, please?

File: packages/flutter_tools/lib/src/dart/pub.dart (hunk 1 above)
Comment (Hixie) [reply packages/flutter_tools/lib/src/dart/pub.dart]: Those docs aren't quite right (e.g. if isRunningOnBot is true but showTraceForErrors is false, we won't trace). Sorry to nitpick...

**Comment by Hixie** [comment]
LGTM modulo minor nitpick.

File: packages/flutter_tools/lib/src/dart/pub.dart (hunk 1 above)
Comment (mit-mit) [reply packages/flutter_tools/lib/src/dart/pub.dart]: Attempted to make them more correct

**Pull request merged**